IBPSA/Fluid/Movers/BaseClasses/Euler/correlation.mo

```diff
@@ -0,0 +1,116 @@
+within IBPSA.Fluid.Movers.BaseClasses.Euler;
+function correlation
+  "Correlation of static efficiency ratio vs log of Euler number ratio"
+  extends Modelica.Icons.Function;
+  input Real x "log10(Eu/Eu_peak)";
+  output Real y "eta/eta_peak";
+
+protected
+  Real a = if x <-0.5 then 0.056873227 else if x>0.5 then -0.0085494313567 else 0.378246;
+  Real b = if x<-0.5 then 0.493231336746 else if x>0.5 then 0.12957001502 else -0.759885;
+  Real c = if x<-0.5 then 1.433531254001 else if x>0.5 then -0.65997315029 else -0.0606145;
+  Real d = if x<-0.5 then 1.407887300933 else if x>0.5 then 1.1399300301 else 1.01427;
+algorithm
+  y:=max(0,a*x^3+b*x^2+c*x+d)/1.01545;
+
+  annotation(smoothOrder = 1,
```

Comment:
This function was designed by @Mathadon, so he would be the best person to ask whether he put special care into these.
But based on some simple computation from my side, the values of dy/dx at the two connecting points are: 0.98295**48** vs 0.98295**50** @ x=-0.5 and -0.5**37** vs -0.5**43** @ x=+0.5.

Reply:
@hcasperfu : Thanks for checking.
@Mathadon : Are these rounding errors -- which are large at x=+0.5 -- or is the implementation not differentiable, which would need to be corrected.

Reply:
@mwetter 
Apologies! I mistyped the numbers. dy/dx @ x=+0.5 is -0.536815**0** vs -0.536815**2**.

Reply:
Thanks for the correction. We should still have higher precision. The current values differ at 1E-6, and solvers sometimes need to solve for higher precision, in some cases, if there are nested loops, this can as well be 1E-10. At that point, this looks like a jump that can throw off algorithms that require continuity of the derivative.

Reply:
I have used higher-precision coefficients provided by @Mathadon for the approximation polynomials in dcd7485.